src/_backend_agg.cpp

```diff
@@ -212,8 +212,8 @@ agg::rect_i RendererAgg::get_content_extents()
 
     r.x1 = std::max(0, r.x1 - 1);
     r.y1 = std::max(0, r.y1 - 1);
-    r.x2 = std::max(r.x2 + 1, (int)width);
-    r.y2 = std::max(r.y2 + 1, (int)height);
+    r.x2 = std::min(r.x2 + 1, (int)width);
+    r.y2 = std::min(r.y2 + 1, (int)height);
```

Comment:
It's zero-based.  The range is like a Python slice, where the lower value is inclusive and the upper value is exclusive.  Hence the need for +1 on the upper values.